ops/charm.py

```diff
@@ -354,6 +354,8 @@ class PreSeriesUpgradeEvent(HookEvent):
     callback method associated with this event, the administrator will initiate
     steps to actually upgrade the series.  After the upgrade has been completed,
     the :class:`PostSeriesUpgradeEvent` will fire.
+
+    .. jujuremoved:: 4.0
```

Comment:
This is nice thinking ahead, but maybe we should wait till 4.0 is (closer to being) shipped.

Reply:
I was thinking that it's better to give charmers as much notice as possible. What if the text was "Scheduled for removal in Juju 4.0" for now, and changed to the simpler "Removed in" once 4.0 makes it out of beta?

Reply:
Fair enough. I like the "Scheduled for removal in Juju 4.0" -- that's nice.